grove/grove_co2_scd30.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env python
+#
+# Library for Grove - CO2 & Temperature & Humidity Sensor for Arduino (SCD30) - 3-in-1
+# (https://www.seeedstudio.com/Grove-CO2-Temperature-Humidity-Sensor-SCD30-p-2911.html)
+#
+
+'''
+## License
+
+The MIT License (MIT)
+
+Copyright (C) 2022  matsujirushi
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+'''
+
+from typing import NoReturn
+from grove.i2c import Bus
+import struct
+import time
+
+
+class GroveCo2Scd30(object):
+
+    def __init__(self, address=0x61, bus=None):
+        self.address = address
+        self.bus = Bus(bus)
+
+        self.set_measurement_interval(2)
+        self.trigger_continuous_measurement()
+
+    @staticmethod
+    def _calc_crc(data: list) -> int:
+        crc = 0xff
+
+        for d in data:
+            crc ^= d
+
+            for _ in range(8):
+                if crc & 0x80:
+                    crc = ((crc << 1) ^ 0x31) & 0xff
+                else:
+                    crc = (crc << 1) & 0xff
+
+        return crc
+
+    def _write(self, cmd: int, data: list):
+        write_data = list(struct.pack(">H", cmd))
+        if data is not None:
+            for d in data:
+                write_data.extend(struct.pack(">H", d))
+                write_data.append(GroveCo2Scd30._calc_crc(struct.pack(">H", d)))
+
+        write_msg = self.bus.msg.write(self.address, write_data)
+        self.bus.i2c_rdwr(write_msg)
+
+    def _read(self, address: int, data_number: int) -> list:
+        write_data = list(struct.pack(">H", address))
+
+        write_msg = self.bus.msg.write(self.address, write_data)
+        self.bus.i2c_rdwr(write_msg)
+
+        time.sleep(0.003)
```

Comment:
Why we need a 3ms delay here?
I check the [Interface Description](https://files.seeedstudio.com/wiki/Grove-CO2-Temperature-Humidity-Sensor-SCD30/res/Sensirion_CO2_Sensors_SCD30_Interface_Description.pdf) of SCD30 and don't find the note that we need this.

Reply:
Removed.